### PR TITLE
Optimization Handling

### PR DESCRIPTION
**Pull request by G-Moris**
Optimization Handling

### CODE REVIEW

File: Client/game_sa/CHandlingManagerSA.cpp [hunk 1]
```diff
@@ -208,102 +216,88 @@ CHandlingManagerSA::~CHandlingManagerSA()
 
 eHandlingProperty CHandlingManagerSA::GetPropertyEnumFromName(std::string strName)
```
Comment (tederis) [inline Client/game_sa/CHandlingManagerSA.cpp]: Pass strName by const reference, not by value.

File: Client/game_sa/CHandlingManagerSA.cpp [hunk 2]
```diff
@@ -9164,10 +9158,10 @@ void CHandlingManagerSA::InitializeDefaultHandlings()
 void CHandlingManagerSA::CheckSuspensionChanges(CHandlingEntry* pEntry)
 {
     // Grab us a multiplayer_sa pointer
-    CMultiplayer* pMultiplayer = g_pCore->GetMultiplayer();
-    eVehicleTypes eModel = pEntry->GetModel();
+    const auto pMultiplayer = g_pCore->GetMultiplayer();
+    const auto eModel = pEntry->GetModel();
```
Comment (tederis) [inline Client/game_sa/CHandlingManagerSA.cpp]: It's better to save the types here

File: Server/mods/deathmatch/logic/CHandlingManager.cpp [hunk 3]
```diff
@@ -160,7 +144,7 @@ bool CHandlingManager::HasModelHandlingChanged(eVehicleTypes eModel)
     if (CVehicleManager::IsValidModel(eModel))
     {
         // Get our Handling ID
-        eHandlingTypes eHandling = GetHandlingID(eModel);
```
Comment (tederis) [inline Server/mods/deathmatch/logic/CHandlingManager.cpp:163]: I'm against using auto in this particular place(and the similar places across PR). This removes important information about a type and forces me to use the search to find out what exactly is covered by `auto`.

File: Server/mods/deathmatch/logic/CHandlingManager.cpp [hunk 4]
```diff
@@ -133,25 +125,17 @@ const CHandlingEntry* CHandlingManager::GetModelHandlingData(eVehicleTypes eMode
...
+        const auto eHandling = GetHandlingID(eModel);
         // Return it
         return m_pModelEntries[eHandling];
     }
-
-    return NULL;
+    return nullptr;
 }
 
 eHandlingProperty CHandlingManager::GetPropertyEnumFromName(std::string strName)
```
Comment (tederis) [inline Server/mods/deathmatch/logic/CHandlingManager.cpp]: Pass strName by const reference, not by value.

**Review by tederis — CHANGES_REQUESTED** [review]

File: Server/mods/deathmatch/logic/CHandlingManager.cpp [hunk 5]
```diff
@@ -169,10 +153,14 @@ bool CHandlingManager::HasModelHandlingChanged(eVehicleTypes eModel)
...
-    // Return if we have changed.
-    m_bModelHandlingChanged[eHandling] = bChanged;
+    // Within range?
+    if (CVehicleManager::IsValidModel(eModel))
+    {
+        // Get our Handling ID
+        const auto eHandling = GetHandlingID(eModel);
+        // Return if we have changed.
+        m_bModelHandlingChanged[eHandling] = bChanged;
+    }
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CHandlingManager.cpp]: Early return

File: Server/mods/deathmatch/logic/CHandlingManager.cpp [hunk 6]
```diff
@@ -133,25 +125,17 @@ const CHandlingEntry* CHandlingManager::GetModelHandlingData(eVehicleTypes eMode
...
     {
         // Get our Handling ID
-        eHandlingTypes eHandling = GetHandlingID(eModel);
+        const auto eHandling = GetHandlingID(eModel);
         // Return it
         return m_pModelEntries[eHandling];
     }
-
-    return NULL;
+    return nullptr;
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CHandlingManager.cpp]: Early return

File: Server/mods/deathmatch/logic/CHandlingManager.cpp [hunk 7]
```diff
@@ -103,13 +98,11 @@ bool CHandlingManager::ApplyHandlingData(eVehicleTypes eModel, CHandlingEntry* p
     if (CVehicleManager::IsValidModel(eModel))
     {
         // Get our Handling ID
-        eHandlingTypes eHandling = GetHandlingID(eModel);
+        const auto eHandling = GetHandlingID(eModel);
         // Apply the data and return success
         m_pModelEntries[eHandling]->ApplyHandlingData(pEntry);
         return true;
     }
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CHandlingManager.cpp]: Early return

File: Server/mods/deathmatch/logic/CHandlingManager.cpp [hunk 8]
```diff
@@ -119,12 +112,11 @@ const CHandlingEntry* CHandlingManager::GetOriginalHandlingData(eVehicleTypes eM
...
     {
         // Get our Handling ID
-        eHandlingTypes eHandling = GetHandlingID(eModel);
+        const auto eHandling = GetHandlingID(eModel);
         // Return it
         return m_pOriginalEntries[eHandling];
     }
-
-    return NULL;
+    return nullptr;
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CHandlingManager.cpp]: Early return

File: Client/game_sa/CHandlingManagerSA.cpp [hunk 9]
```diff
@@ -208,102 +216,88 @@ CHandlingManagerSA::~CHandlingManagerSA()
...
+        // Get our Handling ID, the default value will be HT_LANDSTAL
+        const auto eHandling = GetHandlingID(eModel);
         if (eHandling >= HT_BIKE && eHandling <= HT_FREEWAY)
             return m_pOriginalBikeEntries[eHandling - HT_BIKE];
         else if (eHandling == HT_FAGGIO)
             return m_pOriginalBikeEntries[13];
     }
-
-    return NULL;
+    return nullptr;
```
Comment (TracerDS) [inline Client/game_sa/CHandlingManagerSA.cpp]: Early return

File: Client/game_sa/CHandlingManagerSA.cpp [hunk 10]
```diff
@@ -208,102 +216,88 @@ CHandlingManagerSA::~CHandlingManagerSA()
...
-        if (eHandling < 175 || eHandling > 186)
+        if (eHandling < HT_PREDATOR || eHandling > HT_SEAPLANE)
             return m_pOriginalBoatEntries[0];
         else
-            return m_pOriginalBoatEntries[eHandling - 175];
+            return m_pOriginalBoatEntries[eHandling - HT_PREDATOR];
     }
-
-    return NULL;
+    return nullptr;
```
Comment (TracerDS) [inline Client/game_sa/CHandlingManagerSA.cpp]: Early return

File: Client/game_sa/CHandlingManagerSA.cpp [hunk 11]
```diff
@@ -208,102 +216,88 @@ CHandlingManagerSA::~CHandlingManagerSA()
...
-        if (eHandling < 186 || eHandling > 209)
+        if (eHandling < HT_SEAPLANE || eHandling > HT_RCRAIDER)
             return m_pOriginalFlyingEntries[0];
         else
-            return m_pOriginalFlyingEntries[eHandling - 186];
+            return m_pOriginalFlyingEntries[eHandling - HT_SEAPLANE];
     }
-
-    return NULL;
+    return nullptr;
```
Comment (TracerDS) [inline Client/game_sa/CHandlingManagerSA.cpp]: Early return

File: Client/game_sa/CHandlingManagerSA.cpp [hunk 12]
```diff
@@ -208,102 +216,88 @@ CHandlingManagerSA::~CHandlingManagerSA()
...
-        // Get our Handling ID
-        eHandlingTypes eHandling = GetHandlingID(eModel);
+        // Get our Handling ID, the default value will be HT_LANDSTAL
+        const auto eHandling = GetHandlingID(eModel);
         // Return it
         return m_pOriginalEntries[eHandling];
     }
-
-    return NULL;
+    return nullptr;
```
Comment (TracerDS) [inline Client/game_sa/CHandlingManagerSA.cpp]: Early return

**Review by TracerDS — CHANGES_REQUESTED** [review]

File: Client/game_sa/CHandlingManagerSA.cpp (hunk 12 above)
Comment (G-Moris) [reply Client/game_sa/CHandlingManagerSA.cpp]: Excuse me, by this do you mean that after the condition, immediately make a return?

Comment (TracerDS) [reply Client/game_sa/CHandlingManagerSA.cpp]: yes
```cpp
{
    // Vehicle?
    if (!IsVehicleModel(eModel))
        return nullptr;

    // Get our Handling ID, the default value will be HT_LANDSTAL
    const auto eHandling = GetHandlingID(eModel);
    // Return it
    return m_pOriginalEntries[eHandling];
}
```

**Comment by Dutchman101** [comment]
Because of the team/certain contributor's awareness of specifics surrounding the author's nature, this PR should be reviewed extra carefully one last time, as well for the way that all requested reviews have been implemented. Specifically, there should be attention for any deliberate introduction of weaknesses or suspicious usage of buffers, pointers, etc.

Requesting new reviews from @tederis and @TheNormalnij (if they'd like, of course)

**Comment by G-Moris** [comment]
@Dutchman101 I'm just trying to restore a good relationship with you and help the MTA.

**Comment by Dutchman101** [comment]
Thanks for the PR, it passed the extra scrutiny code reviews and you made all requested adjustments.

> @Dutchman101 I'm just trying to restore a good relationship with you and help the MTA.

I can see you're more serious about that, looking at both this and the anim fixes PR.

But there's still problems that stand in the way of creating a good relationship; you and the other cheat devs from your team need to stop doing malicious things with any findings, and just report them. I will contact you soon (over Discord) with details on what's needed exactly to create a good relationship. @G-Moris 

Anyways, again, thanks for your contributions!

**Comment by G-Moris** [comment]
@Dutchman101 Hello. Why haven't you contacted me in this time? What is the problem?

**Pull request merged**